Fix WordPress app namings on Jetpack app

irfano opened the pull request:
Fixes #17102 

This fixes WordPress naming on the Jetpack app for these screens:
||before|after|
|-|-|-|
|Rating dialog|<img src="https://user-images.githubusercontent.com/2471769/187421378-47bee339-c70e-43a1-bdb7-1a5d534ee074.png" height=640>|<img src="https://user-images.githubusercontent.com/2471769/187509337-f867ee1a-79cb-45ed-a28a-3545738f3a5e.png" height=640>|
|Calendar event|<img src="https://user-images.githubusercontent.com/2471769/187511645-9dd11913-63cc-4710-9626-9e13a6b493a1.png" height=640>|<img src="https://user-images.githubusercontent.com/2471769/187509378-4c6e37a0-50ed-4d67-8db6-f5cb84d87a86.png" height=640>|
|Feature announcement|<img src="https://user-images.githubusercontent.com/2471769/187512153-cf158886-917b-4865-b521-df94b2ac7acc.png" height=640>|<img src="https://user-images.githubusercontent.com/2471769/187510639-ded211da-1f4c-4fd1-b46e-d9367f105f86.png" height=640>|

Note: Screenshots are captured from my development environment. "Jetpack Beta" will be "Jetpack" on the public version.

To test:
**Rating dialog test:**
- Rating dialog logic is a bit complex. There need 10 app launches and 10 interactions (media upload, check notification, publish, open reader post) to show the rating dialog.
Alternatively, you can edit [line 74 of AppRatingDialog](https://github.com/wordpress-mobile/WordPress-Android/blob/e759375f2e177734c4058618c647bd4a302dae91/WordPress/src/main/java/org/wordpress/android/widgets/AppRatingDialog.kt#L74), `return if (shouldShowRateDialog()) {` → `return if (true) {` and build Jetpack app from Android studio to see the rating dialog. Then ensure the title doesn't contain "WordPress".
- Also, test for the WordPress app and ensure the title is "Enjoying WordPress?":.

**Calendar event for scheduling post test:**
1. Launch the Jetpack app.
2. Navigate My Site.
3. Tap the green button at the bottom right of the screen to publish a post.
4. Tap Blog post.
5. Tap "PUBLISH" button.
6. Tap "Publish Date" to schedule the post.
7. Set "Date and Time" to a later date, then Add to calendar button will be available.
8. Tap add to calendar.
9. The calendar app will be opened.
10. Ensure the description of the event doesn't contain WordPress app.
11. Repeat 2-9 for the WordPress app and ensure the app name is WordPress on the description.

**Feature announcement test:**
You need to edit version data and build from Android Studio to see the feature announcement dialog. 
1. Update versionName in [version.properties](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/version.properties) to `79.0`. There is a sample announcement on the announce tool for version 79.0.
2. Launch the Jetpack app.
3. Open Me → App Settings → What's New. The feature announcement dialog will appear. 
4. Ensure the title doesn't contain WordPress.
5. Repeat 3-4 for the WordPress app and ensure the title is "What's New In WordPress".

## Regression Notes
1. Potential unintended areas of impact
Texts on WordPress app.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I tested related screens on the WordPress app.

3. What automated tests I added (or what prevented me from doing so)
No test is added because this is only a change of a word.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

irfano committed:
Fix wrong rating dialog title for Jetpack app

irfano committed:
Fix WordPress naming on calendar event desc for Jetpack app

irfano committed:
Fix WordPress naming on feature announcement for Jetpack app

irfano committed:
Fix failing test of PublishSettingsViewModelTest

ravishanker reviewed (approved):
Thanks for taking care of this 👍 

Feel free to merge it after resolving lint issues

irfano committed:
Remove string resources to fix lint error

The format of "feature_announcement_dialog_label", "calendar_scheduled_post_description", "app_rating_title" changed in previous commits. These are deleted from translation strings files to fix the lint error.

AliSoftware commented:
Nice! ❤️
